ts/@live-compositor/web-wasm/src/wasmInstance.ts

```diff
@@ -0,0 +1,246 @@
+        },
+        [],
+      ];
+    } else if (route.operation === 'update') {
+      return [
+        {
+          type: 'updateScene',
+          outputId: route.id,
+          output: request.body as Api.UpdateOutputRequest,
+        },
+        [],
+      ];
+    }
+  } else if (route.type === 'image') {
+    if (route.operation === 'register') {
+      assert(request.body);
+      return [
+        {
+          type: 'registerImage',
+          imageId: route.id,
+          image: request.body as ImageSpec,
+        },
+        [],
+      ];
+    } else if (route.operation === 'unregister') {
+      return [
+        {
+          type: 'unregisterImage',
+          imageId: route.id,
+        },
+        [],
+      ];
+    }
+  } else if (route.type === 'shader') {
+    throw new Error('Shaders are not supported');
+  } else if (route.type === 'web-renderer') {
+    throw new Error('Web renderers are not supported');
+  }
+
+  throw new Error('Unknown request');
+}
+
+async function handleRegisterOutputRequest(
+  outputId: string,
+  body: RegisterOutputRequest
+): Promise<[WorkerMessage, Transferable[]]> {
+  if (body.type === 'canvas') {
+    const canvas = (body.video.canvas as HTMLCanvasElement).transferControlToOffscreen();
+    return [
+      {
+        type: 'registerOutput',
+        outputId: outputId,
+        output: {
+          type: 'canvas',
+          video: {
+            resolution: body.video.resolution,
+            canvas,
+            initial: body.video.initial,
+          },
+        },
+      },
+      [canvas],
+    ];
+  }
+  throw new Error(`Unknown output type ${body.type}`);
+}
+
+async function handleRegisterInputRequest(
+  inputId: string,
+  body: RegisterInputRequest
+): Promise<[WorkerMessage, Transferable[]]> {
+  if (body.type === 'mp4') {
+    assert(body.url);
+    return [
+      {
+        type: 'registerInput',
+        inputId,
+        input: {
+          type: 'mp4',
+          url: body.url,
+        },
+      },
+      [],
+    ];
+  } else if (body.type === 'camera') {
+    const mediaStream = await navigator.mediaDevices.getUserMedia({
+      audio: false,
+      video: true,
+    });
+    const videoTrack = mediaStream.getVideoTracks()[0];
+    // @ts-ignore
+    const trackProcessor = new MediaStreamTrackProcessor({ track: videoTrack });
+    return [
+      {
+        type: 'registerInput',
+        inputId,
+        input: {
+          type: 'camera',
+          stream: trackProcessor.readable,
+        },
+      },
+      [trackProcessor.readable],
+    ];
+  } else if (body.type === 'screen_capture') {
+    const mediaStream = await navigator.mediaDevices.getDisplayMedia({
+      audio: false,
+      video: {
+        width: { max: 2000 },
+        height: { max: 2000 },
+      },
+    });
+    const videoTrack = mediaStream.getVideoTracks()[0];
+    assert(videoTrack);
+    // @ts-ignore
+    const trackProcessor = new MediaStreamTrackProcessor({ track: videoTrack });
```

Comment:
screenshare is not closed on unregister currently. I'll address that inthe  followup